### PR TITLE
react 4 task

### PR DESCRIPTION
**Pull request by SaXaPhonist**
# Task-4
## Overview
- implement full process of Authorization
 - create forms with validation (use Formik)
 - Use React-query for  api requests
- create mock server
- use redurect if user do not login yet

## [Deploy](https://saxaphonist.github.io/JS-tasks/)

### CODE REVIEW

File: react-notes/src/components/overlayingAlert/overlayingAlert.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import Alert from '@material-ui/lab/Alert';
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+const OverlayingAlert = ({ children, status }) => {
+  const [open, setOpen] = useState(false);
```
Comment (k-krasilnikova) [inline react-notes/src/components/overlayingAlert/overlayingAlert.jsx:6]: What is open? 
Use please `isOpen` instead. The same for the future.
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```

File: react-notes/src/components/overlayingAlert/overlayingAlert.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const OverlayingAlert = ({ children, status }) => {
+  const [open, setOpen] = useState(false);
+  useEffect(() => {
+    setOpen(true);
+  }, []);
+
+  return (
+    open && (
+      <Alert severity={status} onClose={() => setOpen(false)}>
```
Comment (k-krasilnikova) [inline react-notes/src/components/overlayingAlert/overlayingAlert.jsx:13]:
```suggestion
      <Alert severity={status} onClose={setOpen}>
```
And use please prev state inside the `setOpen` function

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]
Amazing!
But some changes required!

File: react-notes/src/components/Forms/loginForm/loginContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import { useDispatch, useSelector } from 'react-redux';
+import * as Yup from 'yup';
+
+import { loginValues, TYPE_LOG, TYPE_USER } from '../../../config/constants';
+import { setCurrUser } from '../../../redux/formRedux/formActions';
+import { getUser } from '../../../restAPI/rest';
+import { closePopup } from '../../../utils/utils';
```
[the remark this answers is not in the document]
Comment (SaXaPhonist) [reply react-notes/src/components/Forms/loginForm/loginContainer.jsx]: have an error when trying to implement absolute paths ( module not found )
 i created a jsconfig.json with such script: 
 ```
{
    "compilerOptions": {
      "baseUrl": "src"
    },
    "include": ["src"]
  }
 ```
but it didn't help me, still the same error

File: react-notes/src/components/Forms/loginForm/loginContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import LoginForm from './loginForm';
+
+const LoginContainer = () => {
+  const isLogout = JSON.parse(localStorage.getItem(TYPE_LOG));
+  const { isAuth } = useSelector((state) => state.form);
+  const [inputValues, setLoginValues] = useState({ ...loginValues });
+  const { email, password } = inputValues;
+  const { data, isSuccess, isError, error } = useQuery([TYPE_USER, email, password], () =>
+    getUser(email, password),
+  );
```
Comment (k-krasilnikova) [inline react-notes/src/components/Forms/loginForm/loginContainer.jsx:20]: Use query should be defined inside api files,not container!

**Review by k-krasilnikova — APPROVED** [review]
Fix please one more comment and it will be available to merge

**Pull request merged**